test/test-generate-pras.jl

```diff
@@ -0,0 +1,241 @@
+        rts_pras_sys,
+        sequential_monte_carlo,
+        PRASInterface.PRAS.Shortfall(),
+    )
+    lole = PRASInterface.PRAS.LOLE(shortfalls)
+    eue = PRASInterface.PRAS.EUE(shortfalls)
+    @test lole isa PRASInterface.PRAS.ReliabilityMetric
+    @test eue isa PRASInterface.PRAS.ReliabilityMetric
+    @test PRASInterface.PRAS.val(lole) >= 0 && PRASInterface.PRAS.val(lole) <= 10
+    @test PRASInterface.PRAS.stderror(lole) >= 0 && PRASInterface.PRAS.stderror(lole) <= 10
+    @test PRASInterface.PRAS.val(eue) >= 0 && PRASInterface.PRAS.val(eue) <= 10
+    @test PRASInterface.PRAS.stderror(eue) >= 0 && PRASInterface.PRAS.stderror(eue) <= 10
+
+    @testset "Lumped Renewable Generators" begin
+        rts_pras_sys =
+            generate_pras_system(rts_da_sys, PSY.Area, lump_region_renewable_gens=true)
+        @test rts_pras_sys isa PRASInterface.PRAS.SystemModel
+        @test test_names_equal(rts_pras_sys.regions.names, area_names)
+
+        rts_pras_sys = generate_pras_system(
+            rts_da_sys,
+            PSY.Area,
+            lump_region_renewable_gens=true,
+            availability=false,
+        )
+        @test rts_pras_sys isa PRASInterface.PRAS.SystemModel
+        @test test_names_equal(rts_pras_sys.regions.names, area_names)
+
+        rts_pras_sys = generate_pras_system(
+            rts_da_sys,
+            PSY.Area,
+            lump_region_renewable_gens=true,
+            availability=false,
+            export_location=joinpath(@__DIR__, "rts.pras"),
+        )
+        @test rts_pras_sys isa PRASInterface.PRAS.SystemModel
+        @test test_names_equal(rts_pras_sys.regions.names, area_names)
+        @test isfile(joinpath(@__DIR__, "rts.pras"))
+        rts_pras_sys2 = PRASInterface.PRAS.SystemModel(joinpath(@__DIR__, "rts.pras"))
+    end
+end
+
+@testset "RTS GMLC with default data" begin
+    rts_da_sys = PSCB.build_system(PSCB.PSISystems, "RTS_GMLC_DA_sys")
+    area_names = PSY.get_name.(PSY.get_components(PSY.Area, rts_da_sys))
+    generator_names =
+        PSY.get_name.(
+            PSY.get_components(
+                x -> PSY.get_available(x) && PSY.get_rating(x) > 0,
+                Union{PSY.HydroDispatch, PSY.RenewableGen, PSY.ThermalGen},
+                rts_da_sys,
+            )
+        )
+    storage_names = PSY.get_name.(PSY.get_components(PSY.Storage, rts_da_sys))
+
+    rts_pras_sys = generate_pras_system(rts_da_sys, PSY.Area)
+    @test rts_pras_sys isa PRASInterface.PRAS.SystemModel
+    @test test_names_equal(rts_pras_sys.regions.names, area_names)
+    @test test_names_equal(rts_pras_sys.generators.names, generator_names)
+    @test test_names_equal(rts_pras_sys.storages.names, storage_names)
+    # 201_HYDRO_4 should have 5.15 FOR, 11.6 POR, and 22 MTTR
+    idx = findfirst(x -> x == "201_HYDRO_4", rts_pras_sys.generators.names)
+    λ, μ = PRASInterface.rate_to_probability(5.15, 22)
+    @test array_all_equal(rts_pras_sys.generators.λ[idx, :], λ)
```

Comment:
Just flagging, this test works for now because we don't have time series data for lambda and mu.

Reply:
I picked a very specific hydro in our "default data" section, so I think this is safe.